libp2p/peerinfo.nim

```diff
@@ -39,6 +39,11 @@ type
       privateKey*: PrivateKey
     of HasPublic:
       key: Option[PublicKey]
+    # gossip 1.1 spec related
+    # https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#explicit-peering-agreements
+    maintain*: bool 
```

Comment:
This shouldn't be part of the`PeerInfo`, this should be moved into the `PubSubPeer`, `PeerInfo` is a libp2p specific type and it should know nothing about pubsub. I also want to avoid any sort of reconnect/monitor functionality in libp2p itself and leave it up to the protocol to decide those flows. The proper way to do it would be to expose lifecycle hooks on the peer/switch that gossip registers and issues reconnects when required.

Reply:
I should add, that since the hooks don't exist right now, to allow the development of this feature to move forward, it is fine for now, but we should rework it before merging. I'll add the hooks ASAP.

Reply:
Oh yeah I agree with you that this was a hack!
Let's see if I can fix it before the final merge, let's keep this unresolved for now.